crates/uv/src/cli.rs

```diff
@@ -485,6 +486,12 @@ pub(crate) struct PipCompileArgs {
     #[arg(long, short)]
     pub(crate) python_version: Option<PythonVersion>,
 
+    /// The platform for which requirements should be resolved.
+    ///
+    /// Represented as a target triple (e.g., `x86_64-pc-windows-msvc`).
```

Comment:
I think it would also be nice to elaborate on what is meant by "target triple" here. Coming from Rust/C/C++/whatever, the notion of the target triple makes intuitive sense. But in Python land, I might know that the things that make up a target are more involved than a standard target triple. I think calling that difference out in the docs will help understanding. And as a concrete example, it might help to answer the question of, "what if I want to specify a different Python implementation?" That isn't captured by the target triple. (And I think the suggestion in that case is to actually use the different Python implementation.)

Maybe a succinct way to explain things here is that the target triples are a heuristic to specify a _part_ of the build parameters, but that they don't capture everything that can be in a "marker environment."

Reply:
Do you think "target triple" is the right abstraction to use?

Reply:
I'm honestly not sure. I am skeptical. With that said, it _does_ seem very useful. I feel like if we present it as a "shortcut" to specifying the most common parts of a marker environment that you might to specify, then it would be useful. (And we can say that there isn't yet a way to specify more parts of the marker environment than what the target triple does.)

I think the weird part of this is that the marker environment derived from the target triple is sort of a frankenstein monster. It's part static information fully determined by the triple and part dynamic information based on the current environment. I don't have enough experience to say how that's going to fall over, but it doesn't seem right to me. For example, target files for rustc don't have any dynamic information in them at all: https://github.com/rust-lang/rust/blob/eff958c59e8c07ba0515e164b825c9001b242294/tests/run-make/target-specs/my-x86_64-unknown-linux-gnu-platform.json

So I wonder if it makes sense to remove the dynamic parts here and, for example, hard-code `cpython`. And for folks that want to use pypy (or whatever), then maybe in the future we can provide a way for end users to specify the complete marker environment.

Reply:
I'm gonna run with it for now... Let's see where it takes us.

> It's part static information fully determined by the triple and part dynamic information based on the current environment.

The only things that are dynamic are the Python-oriented tags: the Python version, the implementation version, etc. So it seems not-horrible to me.